addons/supplies/functions/fnc_traitActions.sqf

```diff
@@ -0,0 +1,71 @@
+#include "script_component.hpp"
+/*
+ * Author: JoramD
+ * Add interactions for traits.
+ *
+ * Arguments:
+ * 0: Object <OBJECT>
+ * 1: Type <STRING>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [object] call tac_supplies_fnc_traitActions
+ *
+ * Public: No
+ */
+
+params ["_object", "_type"];
+
+private _removeTraitsAction = [
+    QGVAR(removeTraits),
+    LLSTRING(RemoveTraits),
+    "",
+    {
+        params ["", "_player"];
+
+        {
+            _player setUnitTrait [_x, false];
+        } forEach ["Medic", "Engineer", "ExplosiveSpecialist", "UavHacker"];
+    },
+    {
+        params ["", "_player"];
+
+        (_player getUnitTrait "Medic") || {_player getUnitTrait "Engineer"};
+    }
+] call ACEFUNC(interact_menu,createAction);
+
+[_object, 0, ["ACE_MainActions"], _removeTraitsAction] call ACEFUNC(interact_menu,addActionToObject);
+
+private _traitAction = [
+    format [QGVAR(%1_Trait), _x],
+    format [LLSTRING(TraitInteraction), _x],
+    "",
+    {
+        (_this select 2) params ["_type"];
+
+        [QGVAR(storeTraits), [getPlayerUID _player, [_type]]] call CBA_fnc_serverEvent;
+
+        private _traits = [_type];
+        if (_type isEqualTo "engineer") then {traits = ["Engineer", "ExplosiveSpecialist", "UavHacker"]};
+        {_player setUnitTrait _x} forEach _traits;
+    },
+    {
+        (_this select 2) params ["_type"];
+
+        private _trainings = [
+            ["getTrainingIdentifiers", _player],
+            EFUNC(apollo,getPlayerInfo),
+            _player,
+            "tac_trainingIdentifiers",
+            60
+        ] call ACEFUNC(common,cachedCall);
+
+        typeOf _player =! "tacs_Unit_B_TeamLeader" && !EGVAR(apollo,enabled) && (if (_type isEqualTo "engineer") then "demolitions" else {_type}) in _trainings;
```

Comment:
```suggestion
        if (_type isEqualTo "engineer") then {
            _type = "demolitions";
        };

        typeOf _player != "tacs_Unit_B_TeamLeader" && {!EGVAR(apollo,enabled)} && {_type in _trainings};
```
We should really just streamline that naming though.

Reply:
Wouldn't this set the `_type` variable to "demolitions" everywhere instead of just this scope?

Reply:
Until the end of this code. Nothing else uses it anymore.

Reply:
It's still used in the statement.

Reply:
Strings are passed as copies, not references.